services/com.mde-network.ep.toolfunctions.xtext/src/main/java/com/mdenetnetwork/ep/toolfunctions/xtext/XtextTool.java

```diff
@@ -69,18 +69,32 @@ public void run(String languageName, String baseName, String extension, String g
 		
 		CliProjectsCreatorMain.main(args);
 		
-		
-		
-		
 		// Add activity files
-		if (grammar != null) {
-			final String grammarSrcPath = languageName.replace('.', '/');
-			final String xtextGrammarPath= PROJECT_PATH + baseName + "/src/" + grammarSrcPath  + ".xtext";
-			recreateGrammarFile(xtextGrammarPath, grammar);
+        //     TODO support multiple file types
+        final String grammarPath = PROJECT_PATH + baseName + "/src/" + baseName.replace('.', '/');
+        final String projectSrcPath = Path.of(grammarPath).getParent().toString();
+        
+        final String shortName = languageName.replace(baseName + ".", "");
+
+		if (grammar != null && !grammar.equals("undefined") ) {
+			final String xtextGrammarPath= grammarPath + "/" + shortName + ".xtext";
+			recreateProjectFile(xtextGrammarPath, grammar);
+		}
+
+		if (validator != null && !validator.equals("undefined") ) {
+			final String xtextValidatorPath= projectSrcPath + "/validation/" + shortName + "Validator.xtend";
+			recreateProjectFile(xtextValidatorPath, validator);		
 		}
 		
-		//TODO add activity files
+		if (scopeProvider != null && !scopeProvider.equals("undefined") ) {
+			final String xtextScopeProviderPath= projectSrcPath + "/scoping/" + shortName + "ScopeProvider.xtend";
+			recreateProjectFile(xtextScopeProviderPath, scopeProvider);
+		}
 		
+		if (generator != null && !generator.equals("undefined") ) {
+			final String xtextGeneratorPath= projectSrcPath + "/generator/" + shortName + "Generator.xtend";
```

Comment:
Shouldn't this be `grammarPath + "/generator/" + shortName + "Generator.xtend"`? The generator usually lives in a sub-directory of the directory containing the `.xtext` file.

Reply:
See https://github.com/mdenet/platformtools/pull/48#discussion_r1471107627 to address in https://github.com/mdenet/platformtools/issues/51